std-bits/table/src/main/java/org/enso/table/data/column/storage/numeric/AbstractLongStorage.java

```diff
@@ -279,6 +279,22 @@ public BoolStorage run(AbstractLongStorage storage) {
                 return new BoolStorage(storage.getIsMissing(), new BitSet(), storage.size(), false);
               }
             })
+        .add(
+            new UnaryMapOperation<>(Storage.Maps.IS_NAN) {
+              @Override
+              public BoolStorage run(AbstractLongStorage storage) {
+                BitSet isNaN = new BitSet(storage.size());
```

Comment:
```suggestion
                BitSet isNaN = new BitSet();
```

I think this will be valid too - IIRC the BitSet will happily answer `false` if `get` index is higher than its size. But then this will avoid allocating an array of size N - so we should get a O(1) storage for this case. Worth trying IMO :)